ci/run_one.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+# Runs shadow's CI for a single configuration.
+# Requires env parameters:
+#   CONTAINER: container-name to use
+#   CC: C compiler to use (cc or clang)
+#   BUILDTYPE: release, debug, or coverage
+#
+# Example:
+#   sudo CONTAINER=centos:8 CC=clang BUILDTYPE=debug ci/run_one.sh
+
+set -euo pipefail
+
+TAG="shadow:${CONTAINER/:/-}-$CC-$BUILDTYPE"
+echo "Running $TAG"
+
+# Build and tag a Docker image witht he given configuration.
+docker build -t "$TAG" -f- . <<EOF
+FROM $CONTAINER
+
+ENV CONTAINER $CONTAINER
+SHELL ["/bin/bash", "-c"]
+
+COPY ci/container_scripts/install_deps.sh /root/install_deps.sh
+RUN . /root/install_deps.sh
+
+ENV CC=$CC BUILDTYPE=$BUILDTYPE
+COPY ci/container_scripts/install_extra_deps.sh /root/install_extra_deps.sh
+RUN . /root/install_extra_deps.sh
+ENV PATH /root/.cargo/bin:\$PATH
+
+RUN mkdir /root/shadow
+COPY . /root/shadow
+WORKDIR /root/shadow
+
+RUN . ci/container_scripts/build_and_install.sh
+EOF
+
+# Run the tests inside the image we just built.
+docker run --shm-size=1g $TAG /bin/bash ci/container_scripts/test.sh
```

Comment:
Should we remove the container here when we're done? For debugging failed tests it might be useful to leave them, but the run_all script will create 29 new containers on each run, so for disk space purposes I think we should delete at least all passing containers.

Reply:
Added a command-line option

Reply:
I'm not clear on how this works. You use `docker run` to make a container based on the image named `$TAG`, but then later it runs `docker rm $TAG`. Since `$TAG` is the image name and you don't assign a name to the container (so it's randomly generated), I think `docker rm` should fail since `$TAG` is not a valid container name?

Reply:
Good catch. Fixed! Per offline discussion, containers are now removed on exit with `--rm`